src/diart/handler.py

```diff
@@ -0,0 +1,309 @@
+        Streaming inference configuration
+    host : str, optional
+        WebSocket server host, by default "127.0.0.1"
+    port : int, optional
+        WebSocket server port, by default 7007
+    key : Union[str, Path], optional
+        SSL key file path for secure WebSocket
+    certificate : Union[str, Path], optional
+        SSL certificate file path for secure WebSocket
+    """
+
+    def __init__(
+        self,
+        config: StreamingHandlerConfig,
+        host: Text = "127.0.0.1",
+        port: int = 7007,
+        key: Optional[Union[Text, Path]] = None,
+        certificate: Optional[Union[Text, Path]] = None,
+    ):
+        self.config = config
+        self.host = host
+        self.port = port
+
+        # Server configuration
+        self.uri = f"{host}:{port}"
+        self._clients: Dict[Text, ClientState] = {}
+
+        # Initialize WebSocket server
+        self.server = WebsocketServer(host, port, key=key, cert=certificate)
+        self._setup_server_handlers()
+
+    def _setup_server_handlers(self) -> None:
+        """Configure WebSocket server event handlers."""
+        self.server.set_fn_new_client(self._on_connect)
+        self.server.set_fn_client_left(self._on_disconnect)
+        self.server.set_fn_message_received(self._on_message_received)
+
+    def _create_client_state(self, client_id: Text) -> ClientState:
+        """Create and initialize state for a new client.
+
+        Parameters
+        ----------
+        client_id : Text
+            Unique client identifier
+
+        Returns
+        -------
+        ClientState
+            Initialized client state object
+        """
+        # Create a new pipeline instance with the same config
+        # This ensures each client has its own state while sharing model weights
+        pipeline = self.config.pipeline_class(self.config.pipeline_config)
+
+        audio_source = src.WebSocketAudioSource(
+            uri=f"{self.uri}:{client_id}",
+            sample_rate=self.config.pipeline_config.sample_rate,
+        )
+
+        inference = StreamingInference(
+            pipeline=pipeline,
+            source=audio_source,
+            batch_size=self.config.batch_size,
+            do_profile=self.config.do_profile,
+            do_plot=self.config.do_plot,
+            show_progress=self.config.show_progress,
+            progress_bar=self.config.progress_bar,
+        )
+
+        return ClientState(audio_source=audio_source, inference=inference)
+
+    def _on_connect(self, client: Dict[Text, Any], server: WebsocketServer) -> None:
+        """Handle new client connection.
+
+        Parameters
+        ----------
+        client : Dict[Text, Any]
+            Client information dictionary
+        server : WebsocketServer
+            WebSocket server instance
+        """
+        client_id = client["id"]
+        logger.info(f"New client connected: {client_id}")
+
+        if client_id not in self._clients:
```

Comment:
To avoid too much nesting, I prefer the pattern:
```python
if client_id in self._clients:
    return

try:
    ...
```

Reply:
this does make things more readable, and explicit. thanks for the note